### DISCUSSION TITLE
Initialize a project vs Update a project

### DISCUSSION THREAD

**Discussion by AljoSt** [projen/projen#1988] [General]
Is there a way to differentiate between creating a new project and updating an existing project? In my project, I have a config file (used for defining machine learning training pipelines). During initialization of the project, I want to create this file and fill it with sensible default options. During the lifecycle of the project, I will then change the config several times, trying out new training-options. If I then want to use projen to update my project, I don't want the config file to be overwritten. Is there a way to accomplish that?

**Comment by mrgrain** [comment]
I think the presence of `project.initProject` can tell you this.
https://projen.io/api/API.html#class-project--
https://projen.io/api/API.html#projen-initproject

**Comment by Chriscbr** [comment]
@AljoSt Perhaps you're looking for `SampleFile`? This file will be generated once, but afterwards is no longer managed / overwritten by projen afterwards. https://projen.io/api/API.html#class-samplefile--

**Comment by AljoSt** [comment]
That looks really good, thank you! Will test it later to see if it's what I need

**Comment by mrgrain** [comment]
@AljoSt FYI I've converted this into a discussion. Feel free to open a new feature request if these options don't suffice, preferably after discussing possible solutions here.
